src/hooks/useIntersectionObserver.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef } from "react";
+
+const useIntersectionObserver = (
+  onIntersect: Function = () => {},
```

Comment:
```suggestion
  onIntersect: () => void,
```

함수 타입의 경우 아무것도 리턴하지않는다면 이런 타입을 사용해주시면 됩니다!